apps/text-extraction/app/schema.py

```diff
@@ -0,0 +1,110 @@
+from typing import List, Optional, Union, Literal
+from dataclasses import dataclass
+
+
+@dataclass
+class CrawlDocument:
+    document_id: str
+    url: str
+    content_type: str
+    blob: bytes
+
+
+@dataclass
+class TextBlock:
+    """
+    Represents a segment of readable text extracted from a document.
+
+    'anchor' is included only if the original HTML element had an 'id' or 'name'.
+    """
+    text: str
+    kind: Literal["body", "caption", "alt", "footnote"] = "body"
+    anchor: Optional[str] = None
+    # dom_path field removed
+    page_number: Optional[int] = None
```

Comment:
is this from a pdf for instance?

Reply:
Yes. Some (but not all) PDF viewers will allow you to jump to a page using a URL fragment.

Reply:
cool